dashboard/app/models/school.rb

```diff
@@ -53,19 +53,34 @@ def full_address
     end.compact.join(' ')
   end
 
+  def most_recent_school_stats
+    school_stats_by_year.order(school_year: :desc).first
+  end
+
   # Determines if this is a high-needs school for the purpose of distributing Maker Toolkit
   # discount codes - this is not a definition we apply broadly.
   # @return [Boolean] True if high-needs, false otherwise.
   def maker_high_needs?
     # As of January 2020, "high-needs" is defined as having >= 50% of the student population
     # eligible for free-and-reduced lunch programs.
-    stats = school_stats_by_year.order(school_year: :desc).first
+    stats = most_recent_school_stats
     if stats.nil? || stats.frl_eligible_total.nil? || stats.students_total.nil?
       return false
     end
     stats.frl_eligible_total.to_f / stats.students_total.to_f >= 0.5
   end
 
+  # determines if school meets Amazon Fugure Engineer criteria
+  # eligible if a) title I school, >40% URM, or >40% eligible for free and reduced meals
+  def afe_high_needs?
+    stats = most_recent_school_stats
+    return false if stats.nil?
+
+    # To align with maker_high_needs? definition above, returning false
+    # if we don't have all data for a given school
```

Comment:
super nit: please end sentences with `.` as it makes it a lot easier to read than when newlines might or might not imply end of sentence. :)